pipelineRunner.go

```diff
@@ -653,7 +666,14 @@ func (pr *pipelineRunnerImpl) tailLogs(ctx context.Context, stageExecutionDone c
 
 			pr.upsertTailLogLine(tailLogLine)
 
-		case <-stageExecutionDone:
+			if tailLogLine.Status != nil && pr.isFinalStageComplete(stages) {
```

Comment:
Any time it receives a status update it checks if the tailing is complete and it can stop.